cfme/tests/physical_infrastructure/test_physical_server_details.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from cfme.utils import testgen
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.physical.provider.lenovo import LenovoProvider
+from cfme.utils.wait import wait_for
+
+pytestmark = [pytest.mark.tier(3)]
+
+pytest_generate_tests = testgen.generate([LenovoProvider], scope="module")
+
+
+@pytest.fixture(scope="module")
+def physical_server(appliance, provider):
+    # Create and then wait for the provider to be added
+    provider.create()
```

Comment:
**Required**
You should be able to use `setup_provider` or `setup_provider_modscope` fixture and remove this.

In `fixtures/provider.py`, this will include control/skipping of problematic providers, will call provider's `setup()` method, and will validate both that the provider appears in CFME, and that the inventory refresh is complete.

You can include the use_fixtures mark in your `pytestmark` list on L9 to cover all the tests in the module.